<a href="https://colab.research.google.com/github/hhammza/Flyrank_ML_Internship_Hamza/blob/main/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Week 01 - Research Question

## 1. My Lane and Why
**Chosen Lane:** Refresh / Content Opportunity Scoring.

I selected the Refresh / Content Opportunity Scoring lane because the internship dataset is designed around helping identify which content pages should be reviewed first. Rather than predicting a business outcome directly, the goal is to rank pages according to their likelihood of needing review based on observable search and engagement signals.

## 2. Research Question
**Question:** Which content pages should be prioritized for review because they show evidence of declining performance while still having enough visibility to justify editorial effort?

**Decision:** Which pages should editors review first?

**Unit of analysis:** One row represents one content page.

**Output:** A ranked list of pages ordered by their estimated refresh priority, along with reason codes explaining each recommendation.

**Action**

A content editor can:

1. refresh outdated pages

2. improve page content

3. update metadata

4. expand thin content

5. monitor pages before further decline



**Cost:**

A false positive means editors spend time reviewing a page that did not need attention.

A false negative means an important page continues losing traffic before anyone notices.

Since editorial resources are limited, prioritization matters more than predicting every declining page.

**Why ML?**

There are many observable signals (CTR, impressions, sessions, freshness, engagement, position, trend, etc.).

Considering these together manually becomes difficult as the number of pages grows.

Machine learning can combine these signals into a ranking that helps humans review the most promising candidates first.

The model supports decision-making rather than replacing human judgment.

## 3. Quick Look at the Data
Replace this text with 2–3 real numbers after running the notebook.

In [1]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("/content/content_refresh_anonymized.csv")

print("="*60)
print("DATASET OVERVIEW")
print("="*60)

print(f"Total Pages: {len(df):,}")
print(f"Total Columns: {len(df.columns)}")

declining_pages = (df["trend_direction"]=="down").sum()

declining_pct = declining_pages / len(df) * 100

high_visibility_declining = df[
    (df["trend_direction"]=="down") &
    (df["impressions_90d"] >= 1000)
].shape[0]

print(f"Declining Pages: {declining_pages:,}")
print(f"Declining Share: {declining_pct:.2f}%")
print(f"Declining Pages with 1000+ Impressions: {high_visibility_declining:,}")

print("\n")

print("Impression Statistics")
display(df["impressions_90d"].describe())

print("\n")

print("Session Statistics")
display(df["sessions_90d"].describe())

DATASET OVERVIEW
Total Pages: 30,000
Total Columns: 44
Declining Pages: 16,262
Declining Share: 54.21%
Declining Pages with 1000+ Impressions: 8,031


Impression Statistics


,impressions_90d
count,30000.000000
mean,5200.366300
std,16838.019547
min,1.000000
25%,81.000000
50%,731.000000
75%,3615.250000
max,517715.000000




Session Statistics


,sessions_90d
count,30000.000000
mean,37.066633
std,107.069131
min,1.000000
25%,2.000000
50%,7.000000
75%,27.000000
max,4345.000000


The starter dataset contains 30,000 content pages collected across 44 variables (features). These pages represent the unit of analysis for this project, where each row corresponds to a single content page.

Approximately 54.21% of all pages are labelled as declining, indicating that a meaningful portion of the content inventory is experiencing reduced performance. Among these, BBBB declining pages have accumulated 1,000 or more impressions during the previous 90 days. These high-visibility pages are particularly important because performance losses on them could have a greater business impact than declines on pages with very little traffic.

The median number of 90-day impressions is 5200, while the median number of 90-day sessions is 37.06. These values indicate that many pages receive measurable search visibility and user traffic, providing sufficient evidence for prioritizing content review rather than relying on random selection.

Overall, these statistics suggest that the dataset contains both declining and healthy pages with varying levels of visibility, making it suitable for developing a machine learning–assisted ranking system that helps content teams decide which pages should be reviewed first.

## 4. Careful Words
This notebook does not prove that refreshing a page will improve its search performance.

Instead, it aims to identify pages that appear to be good candidates for human review based on observable signals.

The model learns associations within historical data rather than causal relationships.

Any recommendation should be reviewed by a content expert before action is taken.

## 5. Self Check

✔ Selected one predefined lane.

✔ Defined the research question.

✔ Defined the decision.

✔ Defined the unit of analysis.

✔ Defined the output.

✔ Explained the action.

✔ Explained the cost of a wrong recommendation.

✔ Explained why ML is useful.

✔ Included real statistics from the starter dataset.

✔ Used careful language and avoided causal claims.

